spec.yaml

```diff
@@ -259,7 +295,9 @@ definitions:
       resourceType:
         type: string
         description: |
-          Resource type (file, calendar, contact,...)
+          Resource type (file, webapp, calendar, contact, ...). A `webapp` share
+          is an application that can be opened on the remote end, typically
+          to enable collaborative workflows.
```

Comment:
This is a (non-binding) idea to further diversify the type of resource one is sharing, including the application engine to use to edit some files.

However, the action of opening a file in a web app would come _after_ the file was already shared with an `/ocm/share` call (and `resourceType=file`), so I'm not entirely sure an additional `/ocm/share` call with `resourceType=webapp` is the right semantic here.

That said, if the two EFSS both use Reva as the reference implementation of OCM, then the `open-in-app` workflow implemented in Reva ought to actually work out of the box, provided that an option (e.g. in the context menu) is given to the user in the web frontend to execute a "remote open" as opposed to a "local open". That hook would be served by the local Reva, which will forward it to the remote Reva as it happens on an OCM share: and the remote Reva will ultimately perform the "remote open", returning the app URL.

Reply:
Resuming the discussion and targeting a resolution at the upcoming workshop @ CERN:

Even assuming that all EFSSs use Reva as the reference implementation of OCM (which is the target model in ScienceMesh), it would still be good to expose an `/ocm/open` endpoint, which Reva would simply forward to `/app/open`. In this context, a `webapp` resource type is indeed redundant, but on the OCM share part some `edit in app` kind of permission may be envisaged to enable such workflow, where the "normal" r/w permission would simply allow to download/upload data to the share.